lib/Service.php

```diff
@@ -230,6 +230,33 @@ function mapValueObject($elementName, $className) {
         $this->classMap[$className] = function(Writer $writer, $valueObject) use ($namespace) {
             return \Sabre\Xml\Serializer\valueObject($writer, $valueObject, $namespace);
         };
+        $this->valueObjectMap[$className] = $elementName;
```

Comment:
Or maybe not. Overwriting *inside* of Writer or Reader makes a ton of sense, but maybe not in Service. Could be good to throw an exception there